src/main/java/uk/gov/hmcts/reform/sscs/reference/data/helper/ReferenceDataHelper.java

```diff
@@ -32,15 +34,35 @@ public static <T> List<T> getReferenceData(String filename, TypeReference<List<T
         }
     }
 
+
+    public static <T extends ReferenceList> Map<String, T> generateHashMapList(Collection<T> collection) {
+        List<Map.Entry<String, T>> entityList = collection.stream()
+                .flatMap(item -> item.getReferenceList().stream()
+                        .map(referenceList -> Map.entry(referenceList.toLowerCase(Locale.ROOT), item)))
+                .collect(Collectors.toList());
+        try {
+            return entityList.stream().collect(Collectors.toMap(Map.Entry::getKey,Map.Entry::getValue));
+        } catch (IllegalStateException exception) {
+            throw new ReferenceDataImportException(String
+                    .format("Reference Data Invalid, there are the following duplicates:%n%s%nError Message: %s",
+                            getDuplicatesString(entityList), exception.getMessage()), exception);
+        }
+
+    }
+
     public static <T> Map<T, T> generateHashMap(Collection<T> collection) {
+        return generateHashMap(collection, reference -> reference);
+    }
+
+    public static <R, T> Map<R, T> generateHashMap(Collection<T> collection,
+                                                   com.google.common.base.Function<T, R> keyFunction) {

```

Comment:
Not really a problem but any reason this includes the package name and the function?